core/ordersync/ordersync.go

```diff
@@ -291,7 +290,7 @@ func (s *Service) GetOrders(ctx context.Context, minPeers int) error {
 
```

Comment:
I think we could definitely benefit from some comments here explaining the purpose of `m`, `wg`, and `semaphore`. Especially the `semaphore` since it is used in a somewhat subtle way. Comments could also help clarify what are we doing in each iteration of the for loop? Under what conditions do we move on to the next iteration or exit the loop?